src/components/UnlockWallet/index.tsx

```diff
@@ -30,6 +31,49 @@ function warnUser(title: string, message?: string | undefined): void {
     message,
   });
 }
+function changeId(chainName : any) {
+  var chains =  JSON.parse(JSON.stringify(internalAddressesPerNetwork))
+  const chain = chains[chainName]
+  try {
+    switch(chain.chainId){
+      // If the chain is default to Metamask if will use wallet_switchEthereumChain
+      case 4:
+        window.ethereum.request({
```

Comment:
Direct use of `window.ethereum` is incompatible with other connectors, such as those from `Frame` or `WalletConnect`. This change will break those, if I understand correctly.

Reply:
This is only supported with Metamask since WalletConnect and Frame can't handle network changes. I replaced the use of window to injectedConnector